drivers: stm32: Make some config symbols menuconfig symbols

erwango opened the pull request:
Move some STM32 drivers  related Kconfig symbols from `config` to `menuconfig` when driver options depends on these symbols.

jfischer-no commented on drivers/usb/device/Kconfig:
```diff
@@ -43,7 +43,7 @@ config USB_DC_RPI_PICO
 DT_STM32_USBHS_SPEED := $(dt_node_str_prop_equals,$(DT_STM32_USBHS),maximum-speed,high-speed)
 
-config USB_DC_STM32
+menuconfig USB_DC_STM32
```
I do not think it makes anything better here, there is only one child and then it is inconsistent.

erwango committed:
drivers: stm32: Make some config symbols menuconfig symbols

Move some STM32 drivers  related Kconfig symbols from `config`
to `menuconfig` when driver options depends on these symbols.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango commented:
- Rebased 
- Adressed @jfischer-no 's comment for USB driver consistency sake
